pkg/registry/thirdpartyresourcedata/codec.go

```diff
@@ -263,6 +263,12 @@ func (t *thirdPartyResourceDataCreator) New(version, kind string) (out runtime.O
 	case "ThirdPartyResourceDataList":
 		return &experimental.ThirdPartyResourceDataList{}, nil
 	default:
-		return t.delegate.New(latest.GroupOrDie("experimental").Version, kind)
+		for _, g := range latest.AllGroups {
```

Comment:
Please plumb group into this function, so we won't have to fix this to have resources of the same name in different groups.